[WIP] PyISPyBClient to be used to read and store data via py-ispyb

IvarsKarpics opened the pull request.

laispc committed:
Fix EPICS command class

- Add missing imports.
- Change relative imports to absolute.
- Get/put command fixes.
- Add reconnect().
- Add auto_monitor as an option.
- Clean up.

laispc committed:
Add configuration folder for beamline lnls_sol

Configuration for development/test at LNLS/Sirius.

laispc committed:
Add configuration folder for beamline lnls_sol

Configuration for development/test at LNLS/Sirius.

laispc committed:
Add configuration folder for beamline lnls_manaca

Configuration for Manacá beamline at LNLS/Sirius.

laispc committed:
Add LNLSMotor class

laispc committed:
Add xml configuration for udiff motors in LNLS/lnls_sol

Simulated EPICS motor PVs for: omega, kappa, phi, x, y, z,
sample x, and sample y.

laispc committed:
Add LNLSCamera class

This class controls a camera connected by EPICS Area Detector.

laispc committed:
Add xml configuration for camera in LNLS/lnls_sol

PVs come from Area Detector simulation IOC.

laispc committed:
Add description to lnls_manaca config README

laispc committed:
Add configuration/lnls_manaca XML files

laispc committed:
Return default value for image sizes in LNLSCamera

Also add info messages about these sizes.

laispc committed:
Remove user idtest0 from inhouse

laispc committed:
WIP: AbstractMachineInfo class

laispc committed:
Add lifetime and topup remaining to AbstractMachineInfo

laispc committed:
Adapt MachInfoMockup to use AbstractMachineInfo

laispc committed:
Remove abstract decorator from some getters in AbstractMachineInfo

laispc committed:
Rename get_values() to get_mach_info_dict()

laispc committed:
Add machine info test

laispc committed:
Fix docstrings

laispc committed:
Run autopep8 and black

laispc committed:
Remove MachineInfoMockup.py

laispc committed:
Rename MachInfoMockup to MachineInfoMockup

laispc committed:
Update PV names of SOL simulated cam config

laispc committed:
Make lnls configuration compatible with latest api changes

- Fix lnls camera script
- md_camera interval set to 100
- Add status flags to LNLSCamera, so log dont get large with errors
- Fix lnls_manaca config xml
- Update xml mockups for lnls_sol
- Using motor mockups until lnls motors is fixed

laispc committed:
WIP: Add var 'stream_hash' to LNLSCamera

laispc committed:
Add class EPICSActuator

laispc committed:
Fix LNLSMotor for new API

laispc committed:
Fix lnls_sol xml config for motors and minidiff

laispc committed:
Add error treatments to LNLSCamera

Image routine can get back to life if, e.g., IOC restarts.

laispc committed:
Add class LNLSZoom

laispc committed:
Add lnls_sol xml for zoom

laispc committed:
Add class LNLSInOut for switches

laispc committed:
Add lnls_sol xml configs for backlight

laispc committed:
Update name in LNLSDiffractometer

move_relative() -> set_value_relative()

laispc committed:
Add lnls_sol xml configs for frontlight

laispc committed:
Add class LNLSCollect

Calls lnls  scan-utils flyscan.

laispc committed:
Add lnls_sol xml for data collection

laispc committed:
Make phiz be focus in lnls_sol config

laispc committed:
Treat infinite limit values in LNLSMotor class

laispc committed:
Fix config of flyscan in LNLSCollect

laispc committed:
Customize session paths for lnls_sol

laispc committed:
Add dir creation in LNLSCollect

laispc committed:
WIP: LNLSBeam

laispc committed:
Add lnls_sol xml for beam

laispc committed:
Fix getters in LNLSCamera

laispc committed:
Fix pre/pos and acquire time args lnls collect

laispc committed:
Add real values for lnls_manaca config

Motors, minidiff, zoom, session, collect, beam info.

laispc committed:
Add LNLSAperture class

laispc committed:
Add aperture xml to lnls_sol

Aperture values list based on Manacá BL case.

laispc committed:
Add LNLSSlits class

laispc committed:
Add slits xml to lnls_sol

Max gap values set to a larger value.

laispc committed:
Configure calib values in LNLSDiffractometer

These values are based on BZoom image calibration.

laispc committed:
Fix default beam position in LNLSBeam

laispc committed:
WIP: Add manual centring routine to LNLSDiff

laispc committed:
Add zoom calibration to LNLSDiffractometer

In LNLSDiff class, add reference to zoom motor, read zoom
calibration from zoom xml, add method to get calibration for
current level and other to update it.

laispc committed:
Add zoom calibration to lnls_sol zoom xml

laispc committed:
Get default beam position on screen from xml in LNLSBeam

laispc committed:
Set back and frontlight as EPICSActuators in lnls_sol

Treating these PVs as actuators (instead of motors) is enough.

laispc committed:
Fix typo in LNLSMotor.set_velocity

velocity -> value

laispc committed:
Add cleanup step to LNLSCollect

At the end of scan, restore omega velocity, which is often modified
by flyscan.

laispc committed:
WIP: LNLS energy and wavelenght

laispc committed:
Move EPICSACtuator and LNLSMotor constants into classes

This makes possible for EPICSActuator and LNLSMotor constants
to be accessed with self.<CONST_NAME>, so their reuse in derivated
classes are easier.

laispc committed:
Add class LNLSDetDistMotor

Also add lnls_sol xml for it.

laispc committed:
Cleanup prints in LNLSInOut

laispc committed:
Add LNLSPilatusDet class

laispc committed:
Add lnls_manaca detector xml

laispc committed:
Check detector threshold in LNLSEnergy

Adapt lnls_sol config xml for this.

laispc committed:
Fix movement check in LNLSDetDistMotor

laispc committed:
Add lnls_manaca dtox xml

laispc committed:
Check error between new and current value in LNLSEnergy

This allows to trigger detector threshold setting only when
needed.

laispc committed:
Add lnls_manaca energy xml

laispc committed:
Calibrate mm_per_pixel separately for x and y

laispc committed:
Add mm_per_pixel_x/y to lnls_manaca zoom xml

laispc committed:
Fix lnls_manaca xmls: aperture beam_info slits

laispc committed:
Configure lnls_manaca xmls: back and frontlights

laispc committed:
Add xy calibration to lnls_sol zoom xml

laispc committed:
Revert lnls_sol energy xml to mock state

laispc committed:
Set pilatus CBF header

laispc committed:
Add LNLSTransmission class and related scripts

laispc committed:
Config lnls_manaca xml: transmission

laispc committed:
Change lnls_manaca/detector beam_xy value

laispc committed:
Remove limit pv in lnls_manaca/dtox

rhfogh commented on HardwareObjects/PyISPyBClient.py:
```diff
@@ -0,0 +1,314 @@
+        if True:
+            #try:
+            if True:
+                session_dict["beamLineName"] = session_dict["beamlineName"]
```
Works fine, but could be a one-liner
 session_dict["beamLineName"] = session_dict.pop("beamlineName")

rhfogh reviewed (approved):
Looks fine, but I am not qualified to judge. Someone from V3 should do that.

beteva commented:
Thanks Ivars

IvarsKarpics committed:
Merge branch 'master' into re_emit_values_fix

beteva committed:
Implement validate_value for a numerical value in AbstractActuator
to avoid duplication in Abstract Motor, Energy and Transmission classes.

beteva committed:
Implement validate_value for a numerical value in AbstractActuator
to avoid duplication in Abstract Motor, Energy and Transmission classes.

beteva committed:
Remove set_wavelength and validate_value (defined in the abstract class).

beteva committed:
Remove set_wavelength and validate_value (defined in the abstract class).

rhfogh committed:
Merge pull request #588 from mxcube/re_emit_values_fix

Rework of re_emit_values

rhfogh committed:
Update guidelines

rhfogh committed:
Bug fix - radius was claculated wrong

rhfogh committed:
Bug fix - radius was claculated wrong

rhfogh committed:
Updating GPhL code

rhfogh committed:
Updating GPhL code

IvarsKarpics committed:
Get xml filename

IvarsKarpics committed:
try to fix travis

IvarsKarpics committed:
try to fix travis 3

IvarsKarpics committed:
try to fix travis 4

IvarsKarpics committed:
try to fix travis 5

IvarsKarpics committed:
try to fix travis 6

IvarsKarpics committed:
try to fix travis 7

IvarsKarpics committed:
try to fix travis 8

IvarsKarpics committed:
try to fix travis 9

IvarsKarpics committed:
try to fix travis 10

IvarsKarpics committed:
revert

IvarsKarpics committed:
Fix travis CI

IvarsKarpics committed:
reverted files

IvarsKarpics committed:
try again 1

IvarsKarpics committed:
try again 2

IvarsKarpics committed:
try again 2

rhfogh committed:
Cleanup - part 1

rhfogh committed:
Cleanup - part 1

marcus-oscarsson committed:
Merge pull request #597 from rhfogh/update_guidelines

Update guidelines

marcus-oscarsson committed:
Merge branch 'master' into xml_path

marcus-oscarsson committed:
Merge pull request #598 from mxcube/xml_path

Get xml filename

marcus-oscarsson committed:
Merge branch 'master' into fix_travis

IvarsKarpics committed:
New try 1

IvarsKarpics committed:
New try 2

IvarsKarpics committed:
New try 3

rhfogh committed:
Addressed issues raised with previous version of PR #600

rhfogh committed:
Addressed issues raised with previous version of PR #600

rhfogh committed:
Merge remote-tracking branch 'rhfogh/snake_case_fix_2' into snake_case_fix_2

# Conflicts:
#	HardwareRepository.py

Commit:
Remove abc decorator from get_mach_info_dict()

Then filling/editing the mach info dict is optional.

Commit:
Improved exception handling

Commit:
Closing fast shutter as early as possible to reduce the time the crystal is exposed

Commit:
Added missing import

Commit:
Cleaned up code

Commit:
Using data_collection_cleanup from ESRFMutliCollect

Commit:
Improved log messages passed to user

Commit:
Added method to recover from detector failure, sometimes simply
taking one image can recover from a previous error

Commit:
Making sure the diffractomter is ready before starting centring

Commit:
Cleaned up code

Commit:
Replaced BaseException with Exception

marcus-oscarsson committed:
Merge pull request #600 from rhfogh/snake_case_fix_2

Snake case fix 2

Commit:
Generating "fake" token for characterization to make it behave like a standard workflow

Commit:
Decreased timeout

Commit:
Merge branch 'master' of https://gitlab.esrf.fr/marcus.oscarsson/HardwareRepository

Conflicts:
	HardwareObjects/EDNACharacterisation.py

marcus-oscarsson committed:
Pinned gevent and greenlet

Pinned gevent and greenlet, fixing a core dump caused by incomparable versions of greenlet and gevent

marcus-oscarsson committed:
Update requirements_python3.txt

marcus-oscarsson committed:
Pinned greenlet for Python 3

IvarsKarpics committed:
Merge branch 'master' into fix_travis

IvarsKarpics committed:
reverted tests

beteva committed:
Merge branch 'master' of github.com:mxcube/HardwareRepository

# Conflicts:
#	HardwareObjects/ESRF/BeamCmds.py
#	HardwareObjects/ESRF/BlissHutchTrigger.py

marcus-oscarsson committed:
Merge pull request #599 from mxcube/fix_travis

Fix travis

Commit:
Fixed issue in Tango poller not updating channel values

Commit:
Adapted to new triggering mode for Eiger

Commit:
Merge branch 'master' of https://gitlab.esrf.fr/marcus.oscarsson/HardwareRepository

Conflicts:
	HardwareObjects/LimaEigerDetector.py

Commit:
Improved state handling

Commit:
Code cleanup

beteva committed:
Merge branch 'issue_587' of github.com:mxcube/HardwareRepository into issue_587

beteva committed:
Merge branch 'master' into issue_587

marcus-oscarsson committed:
Merge pull request #553 from lnls-sol/issue_448_abstract_machine_info_rev2

[CODE CAMP] AbstractMachineInfo issue #448 (Review 2)

IvarsKarpics committed:
Merge branch 'master' into py-ispyb-client

laispc committed:
Add option from_user to set_beam_x

Allow to read beam_x from other sources, e.g., a PV.

laispc committed:
Add option from_user to set_beam_y too

laispc committed:
Add user_beam_* PVs to lnls_manaca detector xml

laispc committed:
Validate Pilatus header setting

Runs collect only if it returned an okay state.

laispc committed:
Simplify pilatus threshold checking in LNLSEnegy

Log some messages more often to make them visible to the user.

laispc committed:
Simplify some sets in LNLSPilatusDet

laispc committed:
Change default beam x in lnls_manaca xml

Commit:
Removed misplaced file

Commit:
improved exception handling

Commit:
Improved messages given to user

Commit:
Fixed problematic import

Commit:
Fixed minor problem with grid result handling

Commit:
Improved exception handling and messages given to user

Commit:
Merge branch 'master' of https://gitlab.esrf.fr/marcus.oscarsson/HardwareRepository

Commit:
More informative message to user when smaple centring is aborted

Commit:
Adding signal for emitting mesh results

laispc committed:
Add set_transmission to LNLSPilatusDet

Call set_transmission in LNLSCollect.
Add det_transmission PV to lnls_manaca detector xml.

Commit:
Making the number of centering rounds configurable

Commit:
Not setting detector cover in from beam commands

Commit:
Moved check for beam logic

Commit:
Increased wait for diffractomter at end of data collection

Commit:
Increased wait for diffractomter at end of data collection

Commit:
Corrected set and get zoom from XML RPC

Commit:
Corrected set and get zoom from XML RPC

Commit:
Added timeout to wait ready before centering and improved messages passed to user

Commit:
Added timeout to wait ready before centering and improved messages passed to user

Commit:
Calling queue_finished_cleanup if collect_hwobj is defined

Commit:
Calling queue_finished_cleanup if collect_hwobj is defined

Commit:
Closing detector cover when changing phase to "BeamLocation",
"Transfer" or "Centring"

Commit:
Closing detector cover when changing phase to "BeamLocation",
"Transfer" or "Centring"

Commit:
Fixed swapped mesh paramters

Commit:
Fixed swapped mesh paramters

Commit:
Fixed swapped mesh paramters

Commit:
Not changing to centering phase after collects

Commit:
Not changing to centering phase after collects

Commit:
Increased centring timeouts

Commit:
Increased centring timeouts

laispc committed:
Set start angle in Pilatus cbf header

Analogous to transmission value setting.

Commit:
Added timeout to wait ready before centering and improved messages passed to user

Commit:
Added timeout to wait ready before centering and improved messages passed to user

Commit:
Merge branch 'master' of https://gitlab.esrf.fr/marcus.oscarsson/HardwareRepository

Conflicts:
	HardwareObjects/Microdiff.py
	HardwareObjects/MiniDiff.py

laispc committed:
Add precision to some comparisons for Pilatus CBF header params

IvarsKarpics committed:
last version

laispc committed:
Set angle increment in Pilatus cbf header

beteva committed:
Fix detector radius if no distance.
Raise RuntimeError if no distance for outer radius.
Fixed bug in force_emit_signals.
Run black. Add documentation.

beteva committed:
Force calculating the wavelength.

beteva committed:
Improve documentation.

beteva committed:
Merge branch 'master' into issue_587

jordiandreu committed:
Missing member in Beamline object

The Qt widget create_discrete_widget requires this run_online_processing boolean value on its initialization.

beteva committed:
Return back to calculate the detector radius with the current distance.

beteva committed:
Calculate the distance in two steps.

marcus-oscarsson committed:
Merge pull request #603 from mxcube/abstract_detector_fix

Detector radius fix

marcus-oscarsson committed:
Merge pull request #606 from mxcube/fix-run_online_processing

Missing member in Beamline object

marcus-oscarsson committed:
Merge branch 'master' into issue_587

laispc committed:
Derive LNLSInOut from AbstractNState

Drop inheritance from Device because it is no longer used by
the new structure of HardwareRepository. AbstractNState is now
used instead to fit the new structure.

Clean up class, removing unused elements.

laispc committed:
Rename LNLSInOut to EPICSNState

As this class is an EPICS layer to NState behavior.
Also:
- Fix EPICSNState.actuatorIn/Out signature.
- Fix xml example on top of file.

laispc committed:
Update lnls_sol xmls to use EPICSNState

laispc committed:
Update lnls_manaca xmls to use EPICSNState

laispc committed:
Clean up LNLSMotor

- Remove override of validate_value(), it is no more necessary.
- Simplify treatment of infinity limits: they are set to (None, None).
- Add xml example on top of file.
- Fix typos.

laispc committed:
Rename LNLSMotor to EPICSMotor

As this class is an EPICS implementation FOR AbstractMotor.

laispc committed:
Update lnls_sol motor xmls to use EPICSMotor

Also, minor fixes in xmls of back/frontlight actuators of this config.

laispc committed:
Update motors xmls of lnls_manaca to use EPICSMotor

Also, minor fixes in back/frontlight xmls.

laispc committed:
Get energy from HWR instead of xml in LNLS classes

- Energy HO is now only defined in beamline_config.yml for any place we use the energy.
- This applies to LNLSCollect and LNLSTransmission.
- Remove ref to energy HO in lnls_manaca xmls for mxcollect and transmission.
- In beamline_config.xml, move energy HO up so it's ready to be used by
other HOs.

laispc committed:
Rename energy xml in lnls_manaca

laispc committed:
Fix names of lnls_manaca XML files

laispc committed:
Fix names of lnls_sol XML files

Commit:
Only adding time argument at init

Commit:
Improved log message

Commit:
Added error handling when reading transfocators

Commit:
Mot moving back to centering phase after queue have finished

Commit:
Using value_to_enum for zoom

Commit:
Added missing " in example

Commit:
Fixed typo

Commit:
Removed unnecessary task decorator

Commit:
Added a way to use the LIMA sequence trigger mode for the Eiger detector

Commit:
Better handling of trigger modes

Commit:
Waiting for detector file names to be set

Commit:
Removed markup from log message

Commit:
Fixes to EMBL sample changer

marcus-oscarsson committed:
Added missing cancel method

marcus-oscarsson committed:
Merge remote-tracking branch 'github/master' into github-master

# Conflicts:
#	BaseHardwareObjects.py
#	Command/Epics.py
#	Command/Pool.py
#	Command/Taco.py
#	Command/Tango.py
#	Command/exporter/MDEvents.py
#	CommandContainer.py
#	HardwareObjectFileParser.py
#	HardwareObjects/ALBA/ALBAPilatus.py
#	HardwareObjects/ALBA/ALBAZoomMotor.py
#	HardwareObjects/Beamline.py
#	HardwareObjects/BlissMotorWPositions.py
#	HardwareObjects/BlissNState.py
#	HardwareObjects/Camera.py
#	HardwareObjects/DESY/NanoDiff.py
#	HardwareObjects/EMBL/EMBLBeamFocusing.py
#	HardwareObjects/EMBL/EMBLDoorInterlock.py
#	HardwareObjects/EMBL/EMBLEnergy.py
#	HardwareObjects/EMBL/EMBLMotorsGroup.py
#	HardwareObjects/EMBL/TINEMotor.py
#	HardwareObjects/ESRF/BeamCmds.py
#	HardwareObjects/ESRF/BlissHutchTrigger.py
#	HardwareObjects/ESRF/ESRFMetadataManagerClient.py
#	HardwareObjects/ESRF/ESRFMultiCollect.py
#	HardwareObjects/ESRF/ESRFSC3.py
#	HardwareObjects/ESRF/ID232MultiCollect.py
#	HardwareObjects/ESRF/ID29XRFSpectrum.py
#	HardwareObjects/ESRF/ID30A3BeamCmds.py
#	HardwareObjects/ESRF/ID30BXRFSpectrum.py
#	HardwareObjects/ESRF/MD2MultiCollect.py
#	HardwareObjects/EdnaWorkflow.py
#	HardwareObjects/GenericDiffractometer.py
#	HardwareObjects/LimaEigerDetector.py
#	HardwareObjects/MAXIV/BIOMAXCollect.py
#	HardwareObjects/MAXIV/BIOMAXMD3.py
#	HardwareObjects/MAXIV/BIOMAXMD3Camera.py
#	HardwareObjects/MAXIV/MaxIVSession.py
#	HardwareObjects/MachCurrent.py
#	HardwareObjects/MicrodiffInOut.py
#	HardwareObjects/MiniDiff.py
#	HardwareObjects/MotorWPositions.py
#	HardwareObjects/MultiplePositions.py
#	HardwareObjects/QtGraphicsManager.py
#	HardwareObjects/QueueManager.py
#	HardwareObjects/RobodiffMotorWPositions.py
#	HardwareObjects/SOLEIL/PX1/PX1Attenuator.py
#	HardwareObjects/SOLEIL/PX1/PX1Energy.py
#	HardwareObjects/SOLEIL/PX1/PX1TangoLight.py
#	HardwareObjects/SOLEIL/PX2/PX2Attenuator.py
#	HardwareObjects/SOLEIL/PX2/PX2Guillotine.py
#	HardwareObjects/SOLEIL/SOLEILEnergyScan.py
#	HardwareObjects/SOLEIL/SOLEILPss.py
#	HardwareObjects/SardanaMotor.py
#	HardwareObjects/SpecMotorWPositions.py
#	HardwareObjects/TangoLimaMpegVideo.py
#	HardwareObjects/XMLRPCServer.py
#	HardwareObjects/XRFSpectrum.py
#	HardwareObjects/abstract/AbstractAperture.py
#	HardwareObjects/abstract/AbstractVideoDevice.py
#	HardwareObjects/mockup/ApertureMockup.py
#	HardwareObjects/mockup/BeamlineTestMockup.py
#	HardwareObjects/mockup/MicrodiffInOutMockup.py
#	HardwareObjects/sample_centring.py
#	HardwareRepository.py
#	Poller.py
#	deprecated/HardwareObjects/NamedState.py
#	deprecated/HardwareObjects/Server/__init__.py

IvarsKarpics committed:
Merge pull request #608 from lnls-sol/lnls-site

LNLS site-specific config

marcus-oscarsson committed:
Merge branch 'master' into esrf-updates

marcus-oscarsson committed:
Only moving the detector cover up if its down

marcus-oscarsson committed:
camel to snake case

marcus-oscarsson committed:
Absolute import

marcus-oscarsson committed:
Camel to snake case

beteva committed:
Merge branch 'master' into issue_587

IvarsKarpics committed:
Merge pull request #610 from mxcube/esrf-updates

Camel to snake case updates for latest local changes

beteva committed:
Merge branch 'master' into issue_587

beteva committed:
Remove _wavelength_value. Always calculate the wavelength from the energy.

beteva committed:
Add the use of the value to check if action finished.

beteva committed:
Remove unused method.

beteva committed:
Cleanup, black.

marcus-oscarsson committed:
Merge pull request #596 from mxcube/issue_587

validate_value in AbstractActuator

marcus-oscarsson committed:
Update README.md

marcus-oscarsson committed:
Update CONTRIBUTING.md

IvarsKarpics committed:
Merge branch 'master' into py-ispyb-client

IvarsKarpics commented:
It is difficult to fix the pr and I am closing this PR and will open a new one
